py/tests/integration/test_spend_tx.py

```diff
@@ -88,6 +88,83 @@ def test_successful():
     common.stop_node(alice_node)
     shutil.rmtree(root_dir)
 
+def test_not_enough_tokens():
+    # Bob should not be able to send more tokens than he has
+    #
+    # Let's say Bob has 100 tokens. He should not be able to send more than
+    # 100 tokens to Alice.
+    #
+    # If there's an incoming but unconfirmed deposit into Bob's account then Bob
+    # should not be able to use the incoming tokens until the spend transaction
+    # they are in is confirmed.
+    test_settings = settings["test_not_enough_tokens"]
+    coinbase_reward = common.coinbase_reward() 
+    (root_dir, bob_node, bob_api, bob_top) = setup_node_with_tokens(test_settings, "bob") 
+
+    spend_tx_amt = test_settings["spend_tx"]["amount"]
+    spend_tx_fee = test_settings["spend_tx"]["fee"]
+    bob_balance = bob_api.get_account_balance()
+    bob_has_not_enough_tokens = bob_balance.balance < spend_tx_amt + spend_tx_fee
+    assert_equals(bob_has_not_enough_tokens, True)
+    print("Bob initial balance is " + str(bob_balance.balance) +
+            " and he will try to spend " + str(spend_tx_amt + spend_tx_fee))
+
+    alice_balance0 = {"balance":0} 
+    try:
+        alice_balance0 = bob_api.get_account_balance(pub_key=test_settings["spend_tx"]["alice_pubkey"])
+    except ApiException as e:
+        assert_equals(e.status, 404) # Alice has no account yet
+
+    # Bob tries to send some tokens to Alice
+    bob_internal_api = common.internal_api(bob_node)
+    spend_tx_obj = SpendTx(
+        recipient_pubkey=test_settings["spend_tx"]["alice_pubkey"],
+        amount=spend_tx_amt,
+        fee=spend_tx_fee)
+    print("Bob's spend_tx is " + str(spend_tx_obj))
+    bob_internal_api.post_spend_tx(spend_tx_obj)
+    print("Transaction sent")
+
+    # ensure Bob balance had not changed
+    bob_balance1 = bob_api.get_account_balance()
+    assert_equals(bob_balance.balance, bob_balance1.balance)
+
+    # ensure Alice balance had not changed
+    try:
```

Comment:
Fragile. Same as https://github.com/aeternity/epoch/pull/383/files#r152546842

If correct, please track this in a ticket so that we do not lose track of it.

Reply:
Same comment as above: in this test case Bob's node mines and Alice's balance shouldn't change